### PR TITLE
Changes to support SC mode cluster

### PR DESCRIPTION
**Pull request by sud82**

### CODE REVIEW

File: deployment/strong_consistency.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+				return err
+			}
+
+			if err := setFilteredRosterNodes(clHost, scNs, rosterNodes, rosterBlockList); err != nil {
+				return err
+			}
+		}
+	}
+
+	return runRecluster(clHosts)
```
Comment (abhishekdwivedi3060) [inline deployment/strong_consistency.go:60]: Can we add a check here to run `runRecluster` only when atleast one of the namespace's roster is actually set when diff found. Skip it when roster was same for all ns across all nodes

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

File: deployment/strong_consistency.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+var (
+	rosterKeyObservedNodes     = "observed_nodes"
+	rosterKeyRosterNodes       = "roster"
+	nsKeyUnavailablePartitions = "unavailable_partitions"
+	nsKeyDeadPartitions        = "dead_partitions"
+	nsKeyStrongConsistency     = "strong-consistency"
+)
+
+func GetAndSetRoster(log logr.Logger, hostConns []*HostConn, policy *as.ClientPolicy, rosterBlockList []string, removedNamespaces []string) error {
+	log.Info("Check if we need to Get and set roster for SC namespaces")
```
Comment (tanmayja) [inline deployment/strong_consistency.go:21]: nit pick: either make "get and set" or "Get and Set"

File: deployment/strong_consistency.go [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+		return false, err
+	}
+	clHost.log.Info("Check if namespace is SC enabled", "ns", ns, nsKeyStrongConsistency, scBool)
+
+	return scBool, nil
+}
+
+// Info calls
+
+func recluster(clHost *host) error {
```
Comment (tanmayja) [inline deployment/strong_consistency.go:208]: can we use this recluster function in other places, where recluster command is already being called.

File: deployment/strong_consistency.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+		if strings.EqualFold(ele, listEle) {
+			return true
+		}
+	}
+	return false
+}
+
+//----------------------------------------------------------------
+
+func isNodeInRoster(clHost *host, ns string) (bool, error) {
```
Comment (tanmayja) [inline deployment/strong_consistency.go:298]: ARE we using this function anywhere?

Comment (sud82) [reply deployment/strong_consistency.go:298]: Removed

File: deployment/strong_consistency.go (hunk 3 above)
Comment (sud82) [reply deployment/strong_consistency.go:214]: Right now these two recluster call implementations are a bit different. I feel that some older functions need rework. I will try to do that reformatting in a different commit.

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Pull request merged**